presto-spi/src/main/java/com/facebook/presto/spi/plan/JoinType.java

```diff
@@ -20,7 +20,8 @@ public enum JoinType
     INNER("InnerJoin"),
     LEFT("LeftJoin"),
     RIGHT("RightJoin"),
-    FULL("FullJoin");
+    FULL("FullJoin"),
+    SOURCE_OUTER("SourceOuter");
```

Comment:
What does SOURCE_OUTER mean? Left join?

Reply:
I haven't got the chance to dig deep in the execution. Based on the code, if it's `SOURCE_OUTER`, Presto execution planner will create a LookupJoinOperator with JoinType PROBE_OUTER (the Probe is the outer side of the join). (https://github.com/prestodb/presto/blob/47a708022aba003ebfda483ea3c940231fdbd282/presto-main-base/src/main/java/com/facebook/presto/operator/LookupJoinOperators.java#L99-L110)